### PR TITLE
MES-2901 - Get view test result ready

### PR DESCRIPTION
**Pull request by TomBillingtonUK**
## Description

- Creates 2 new components - Data Row & Data Row Custom
- Refactors the View Test Result Cards to use the new components

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/view-test-result/cat-b/components/vehicle-details-card/vehicle-details-card.html [hunk 1]
```diff
@@ -4,39 +4,31 @@ <h4>Vehicle details</h4>
...
+        [label]="'Transmission'"
+        [value]="data.transmission"
+        *ngIf="data.transmission"
+        [shouldHaveSeperator]="data.registrationNumber || data.vehicleDetails || data.instructorRegistrationNumber  ? true: false">
+      </data-row>
+      <data-row
+        [label]="'Vehicle registration number'"
+        [value]="data.registrationNumber"
+        *ngIf="data.registrationNumber"
+        [shouldHaveSeperator]="data.vehicleDetails ? true: false">
```
Comment (ammarhaiderbjss) [inline src/pages/view-test-result/cat-b/components/vehicle-details-card/vehicle-details-card.html:17]: space before the `:` please

File: src/pages/view-test-result/cat-b/components/vehicle-details-card/vehicle-details-card.html [hunk 2]
```diff
@@ -4,39 +4,31 @@ <h4>Vehicle details</h4>
...
+      </data-row>
+      <data-row-custom
+        [label]="'Vehicle details'"
+        *ngIf="data.vehicleDetails"
+        [shouldHaveSeperator]="data.instructorRegistrationNumber ? true: false">
```
Comment (ammarhaiderbjss) [inline src/pages/view-test-result/cat-b/components/vehicle-details-card/vehicle-details-card.html]: space before the `:` please

**Review by rstepnowski — APPROVED** [review]

File: src/pages/view-test-result/cat-b/components/test-summary-card/test-summary-card.html [hunk 3]
```diff
@@ -4,91 +4,35 @@ <h4>Test summary</h4>
...
+        [label]="'Licence provided'"
+        [value]="getConvertBooleanToString(data.provisionalLicenceProvided)"
+        *ngIf="shouldDisplayLicenceProvided(data.provisionalLicenceProvided)">
+      </data-row>
+      <data-row [label]="'Certificate number'" [value]="data.passCertificateNumber" *ngIf="data.passCertificateNumber"></data-row>
+      <data-row-custom [label]="'Route number'">
+        <span class="mes-data" *ngIf="!data.routeNumber">None</span>
+        <span class="mes-data">{{ data.routeNumber }}</span>
+      </data-row-custom>
+      <data-row-custom [label]="'Independent driving'">
```
Comment (ammarhaiderbjss) [inline src/pages/view-test-result/cat-b/components/test-summary-card/test-summary-card.html:25]: Lots of these strings, could we perhaps use existing enums or create new ones?

File: src/pages/view-test-result/cat-b/components/test-summary-card/test-summary-card.html [hunk 4]
```diff
@@ -4,91 +4,35 @@ <h4>Test summary</h4>
...
-          <span class="mes-data">{{ data.additionalInformation }}</span>
-        </ion-col>
-      </ion-row>
+      </data-row-custom>
+      <data-row
+        [label]="'Licence provided'"
+        [value]="getConvertBooleanToString(data.provisionalLicenceProvided)"
+        *ngIf="shouldDisplayLicenceProvided(data.provisionalLicenceProvided)">
+      </data-row>
+      <data-row [label]="'Certificate number'" [value]="data.passCertificateNumber" *ngIf="data.passCertificateNumber"></data-row>
```
Comment (ammarhaiderbjss) [inline src/pages/view-test-result/cat-b/components/test-summary-card/test-summary-card.html]: can we format these attributes to be on individual lines to be consistent

**Review by ammarhaiderbjss — APPROVED** [review]

File: src/pages/view-test-result/cat-b/components/vehicle-details-card/vehicle-details-card.html (hunk 1 above)
Comment (TomBillingtonUK) [reply src/pages/view-test-result/cat-b/components/vehicle-details-card/vehicle-details-card.html:17]: done

File: src/pages/view-test-result/cat-b/components/vehicle-details-card/vehicle-details-card.html (hunk 2 above)
Comment (TomBillingtonUK) [reply src/pages/view-test-result/cat-b/components/vehicle-details-card/vehicle-details-card.html]: done

File: src/pages/view-test-result/cat-b/components/test-summary-card/test-summary-card.html (hunk 3 above)
Comment (TomBillingtonUK) [reply src/pages/view-test-result/cat-b/components/test-summary-card/test-summary-card.html:25]: As they are only labels happy to do this as a new PR - we could also do it across the app :)

File: src/pages/view-test-result/cat-b/components/test-summary-card/test-summary-card.html (hunk 4 above)
Comment (TomBillingtonUK) [reply src/pages/view-test-result/cat-b/components/test-summary-card/test-summary-card.html]: done

**Pull request merged**